src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,157 @@
+package duke;
+
+import java.io.IOException;
+
+import javafx.application.Application;
+import javafx.fxml.FXML;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+/**
+ * Duke bot object class, which has a task list, a storage space and a user interface.
+ */
+public class Duke extends Application {
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+
+    private Image userImg = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image dukeImg = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+    /**
+     * Creates a Duke object with the given filepath.
+     */
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage("/data/tasks.txt");
+        TaskList currentTasks;

```

Comment:
Similar to the another comment, perhaps this could also be singular since TaskList is technically one object which is a list of tasks? 
```suggestion
        TaskList currentTaskList;

```

Reply:
Thanks for highlighting!